### PR TITLE
SOLR-16941: Use smarter default Solr URL in CLI

### PR DESCRIPTION
**Pull request by HoustonPutman**
https://issues.apache.org/jira/browse/SOLR-16941

I've also updated one of the SSL tests to show that this works (defaulting the scheme at least)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cli/SolrCLI.java [hunk 1]
```diff
@@ -97,13 +97,13 @@ public class SolrCLI implements CLIO {
...
               "Base Solr URL, which can be used to determine the zkHost if that's not known; defaults to: "
-                  + DEFAULT_SOLR_URL
+                  + getDefaultSolrUrl()
                   + '.')
           .build();
   public static final Option OPTION_VERBOSE =
       Option.builder("verbose").required(false).desc("Enable more verbose command output.").build();
 
-  // should this be boolean or just a otption?
+  // should this be boolean or just an option?
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/cli/SolrCLI.java:106]: thank you for fixing this typo...  it' bugged me for a while.

File: solr/packaging/test/test_ssl.bats [hunk 2]
```diff
@@ -54,8 +54,8 @@ teardown() {
   solr start -c
   solr assert --started https://localhost:8983/solr --timeout 5000
 
-  run curl --cacert "$ssl_dir/solr-ssl.pem" 'https://localhost:8983/solr/admin/collections?action=CREATE&collection.configName=_default&name=test&numShards=2&replicationFactor=1&router.name=compositeId&wt=json'
-  assert_output --partial '"status":0'
+  run solr create -c test -s 2
```
Comment (epugh) [inline solr/packaging/test/test_ssl.bats:57]: Did you mean this change?    I guess this is demonstrating that bin/solr create works with https?    When you define those variables?

**Review by epugh — COMMENTED** [review]
LGTM.   This just touches bin/solr, what about bin/solr.cmd?

File: solr/packaging/test/test_ssl.bats (hunk 2 above)
Comment (epugh) [reply solr/packaging/test/test_ssl.bats:57]: I looked, and I see another curl example elsewhere in the tests.  I actually like the example of using curl with the `solr-ssl.pem`, it helped me understand how to interact with Solr in SSL ;-)

Comment (HoustonPutman) [reply solr/packaging/test/test_ssl.bats:57]: Yep, it's to test the usage!

**Comment by HoustonPutman** [comment]
> This just touches bin/solr, what about bin/solr.cmd?

Not exactly sure but from my surface level research, the windows variables do not need to be exported like they do in bash, so there's nothing to change. The bash changes are just to export the variables we use in the CLI.

**Review by epugh — APPROVED** [review]

**Comment by HoustonPutman** [comment]
Ok fixed the issues. @epugh Going forward (10.0) the `package` tool should probably work like the rest of the tools, with regards to the Solr Port.

**Pull request merged**